tests/unit/test_http_session.py

```diff
@@ -367,6 +367,20 @@ def test_proxy_ssl_context_uses_check_hostname(self):
             last_call = self.proxy_manager_fun.call_args[-1]
             self.assertIs(last_call['ssl_context'].check_hostname, True)
 
+    def test_proxy_ssl_context_does_not_use_check_hostname_if_ip_address(self):
```

Comment:
Let's make sure we're testing this for both IPv4 and IPv6 proxies.

Reply:
Thanks @nateprewitt, that makes sense.
But now when I tried testing IPv6 here, I am seeing a problem. So, I used the IPv6 address `https://[2001:0db8:85a3:0000:0000:8a2e:0370:7334]:5678` which was one of the examples in [urllib3](https://github.com/urllib3/urllib3/blob/fab33936db9281e96b6325b4fafe2fbb852fa458/test/test_ssl.py#L18) to test is_ipaddress. I had to use the braces format here so that I can pass in the port (not passing the port get `parse_url` to fail.). Now, the `is_ipaddress` that we use here from [urllib3](https://github.com/urllib3/urllib3/blob/fab33936db9281e96b6325b4fafe2fbb852fa458/src/urllib3/util/ssl_.py#L571) only matches for `_BRACELESS_IPV6_ADDRZ_RE`. is it that hostname at that point is expected to be without braces?

Reply:
Using `IPV6_ADDRZ_RE` instead, works.
Also, not having a port name doesn't create a problem when using braces.

Reply:
Thanks for bringing this to my attention, @vabhasin! I'm currently discussing options with the other urllib3 maintainers. I think ideally we want to upstream this so we're not maintaining it in two places. That will unfortunately not be in all of our supported versions, so we'll need a temporary wrapper.

I'll follow up shortly with more information on how we want to proceed. I think the current vending isn't going to be the path forward though.